packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -56,54 +65,135 @@ export const Skeleton = () => {
+    )
+
+    // NOTE: variationAttributes are only defined for master/variant type products.
+    const variationAttributes = useMemo(() => getDecoratedVariationAttributes(product), [product])
 
     // ProductTile is used by two components, RecommendedProducts and ProductList.
     // RecommendedProducts provides a localized product name as `name` and non-localized product
     // name as `productName`. ProductList provides a localized name as `productName` and does not
     // use the `name` property.
     const localizedProductName = product.name ?? product.productName
 
-    const {currency: activeCurrency} = useCurrency()
-    const isFavouriteLoading = useRef(false)
-    const styles = useMultiStyleConfig('ProductTile')
-
     return (
         <Box {...styles.container}>
-            <Link
-                data-testid="product-tile"
-                to={productUrlBuilder({id: productId}, intl.local)}
-                {...styles.link}
-                {...rest}
-            >
+            <Link data-testid="product-tile" to={productUrl} {...styles.link} {...rest}>
                 <Box {...styles.imageWrapper}>
-                    {image && (
-                        <AspectRatio {...styles.image}>
-                            <DynamicImage
-                                src={`${image.disBaseLink || image.link}[?sw={width}&q=60]`}
-                                widths={dynamicImageProps?.widths}
-                                imageProps={{
-                                    // treat img as a decorative item, we don't need to pass `image.alt`
-                                    // since it is the same as product name
-                                    // which can cause confusion for individuals who uses screen readers
-                                    alt: '',
-                                    ...dynamicImageProps?.imageProps
-                                }}
-                            />
-                        </AspectRatio>
-                    )}
+                    <AspectRatio {...styles.image}>
+                        <DynamicImage
+                            src={`${
+                                image?.disBaseLink ||
+                                image?.link ||
+                                product?.image?.disBaseLink ||
+                                product?.image?.disBaseLink
+                            }[?sw={width}&q=60]`}
+                            widths={dynamicImageProps?.widths}
+                            imageProps={{
+                                // treat img as a decorative item, we don't need to pass `image.alt`
+                                // since it is the same as product name
+                                // which can cause confusion for individuals who uses screen readers
+                                alt: '',
+                                ...dynamicImageProps?.imageProps
+                            }}
```

Comment:
We have an opportunity to improve performance here, by lazy loading the image further (via native img attribute). In this case, if it's only above the page fold, then it will be downloaded.

```suggestion
                            imageProps={{
                                // treat img as a decorative item, we don't need to pass `image.alt`
                                // since it is the same as product name
                                // which can cause confusion for individuals who uses screen readers
                                alt: '',
                                loading: 'lazy',
                                ...dynamicImageProps?.imageProps
                            }}
```

Reply:
Always room for improvement 👍